From b5b5fa43355b432414f336a1795f49709d2dd388 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 26 Aug 2016 03:28:26 -0400
Subject: [PATCH 50/62] support to get kickstart from network

We support http/https/ftp protocol to get kickstart file

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 anaconda | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/anaconda b/anaconda
index 4b65adc..94e1268 100755
--- a/anaconda
+++ b/anaconda
@@ -1066,8 +1066,12 @@ if __name__ == "__main__":
         ksFiles = ["/tmp/updates/interactive-defaults.ks",
                    "/usr/share/anaconda/interactive-defaults.ks"]
 
+    log.info("ksFiles: %s" % ksFiles)
     for ks in ksFiles:
-        if not os.path.exists(ks):
+        if not os.path.exists(ks) and \
+           not ks.startswith("http://") and \
+           not ks.startswith("https://") and \
+           not ks.startswith("ftp://"):
             continue
 
         kickstart.preScriptPass(ks)
-- 
2.7.4

